xhttp/identity/ctx.go

```diff
@@ -69,6 +69,12 @@ func SetGlobalIdentityMapper(e Mapper) {
 	identityMapper = e
 }
 
+//FromContext extracts the RequestContext stored inside a go context. Returns null if no such value exists.
+func FromContext(ctx context.Context) *RequestContext {
+	ret, _ := ctx.Value(keyContext).(*RequestContext)
```

Comment:
qq,,,  why do we ignore second parameter? Also is this line prone to "nil pointer exception"?

Reply:
Ah ok ,, just read the comment. Presumably we just plan to return null by default and leave it to the caller to check if return  value is null.

Reply:
There are 2 forms of dynamic casts in golang:

```
ret = orig.(Type)
```

and

```
ret, ok = orig.(Type)
```

The first one will panic if orig cannot be dynamically cast to `Type`. The second one will return `ok == false` instead of panic. 

We want the second one, but we don't need the ok variable because if `ok == false` then `ret == <zero value>`.

Anyways, this has the behavior of "Extract a request context out of a go context, if set, and return nil otherwise".